### PR TITLE
Add autofix for `D300`

### PR DESCRIPTION
**Pull request by diceroll123**
## Summary

Add fix for `D300`

## Test Plan

`cargo test` and manually

### CODE REVIEW

File: crates/ruff_linter/src/rules/pydocstyle/rules/triple_quotes.rs [hunk 1]
```diff
@@ -60,18 +73,28 @@ pub(crate) fn triple_quotes(checker: &mut Checker, docstring: &Docstring) {
     match expected_quote {
         Quote::Single => {
             if !leading_quote.ends_with("'''") {
-                checker.diagnostics.push(Diagnostic::new(
-                    TripleSingleQuotes { expected_quote },
+                let mut diagnostic =
+                    Diagnostic::new(TripleSingleQuotes { expected_quote }, docstring.range());
+
+                diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
+                    prefixes + &format!("'''{}'''", docstring.body().as_str()),
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/pydocstyle/rules/triple_quotes.rs:80]: Is this always safe? Could this ever change the inner contents of the string?

Comment (diceroll123) [reply crates/ruff_linter/src/rules/pydocstyle/rules/triple_quotes.rs]: ran it on the cpython source with `--fix`, looks fine, but I trust your judgement!

Comment (diceroll123) [reply crates/ruff_linter/src/rules/pydocstyle/rules/triple_quotes.rs]: I stand corrected, it breaks when confronted with this condition:

```
'''Ensure that the given "actual" string ends with "exp_end"'''
```

**Comment by diceroll123** [comment]
_Didn't want to commit with your automerge on, just in case_ 😅

**Pull request merged**